datatypes/src/raster/typed_raster.rs

```diff
@@ -203,3 +366,113 @@ where
         }
     }
 }
+
+pub trait TypedRasterConversion {
+    fn get_raster(raster: TypedRaster2D) -> Option<Raster2D<Self>>
+    where
+        Self: Pixel;
+
+    fn get_typed_raster(_raster: Raster2D<Self>) -> TypedRaster2D
+    where
+        Self: Pixel,
+    {
+        unimplemented!()
+    }
```

Comment:
Do we need this block? This method exists on the trait anyway, but the compiler won't warn us to implement it.

Reply:
yes you should remove the default implementation and add a dimension.

Reply:
0c148c7dc15557c77a7dd6d7f9cd11c43545bfcd I did not add a dimension though as we do not support any operations on 3d rasters anyway